### ISSUE TITLE
Pretraining matters？

### ISSUE THREAD

**Issue by morgenzhang** [heshuting555/TransReID#8]
Many thanks for you breakthrough work of introducing transformer into Person Re-id. I have tried your code locally and it works
 well as expected. However, I just find that the pretraining models from imagenet may have a significant influence on the downstream reid performance. The vit-base model was pretrained on imagenet21k and finetuned on imagenet 1k afterwards, while other cnn-based networks were mostly pretrained on imagenet 1k directly instead. So in fair, I think we have to compare these models by using the pretrained model from the same data scale.
I have also replaced the vit with the swin architecture which provides 1k pretrained model as well as 22k. The experiments with 22k pretrained models on market1501 and msmt17 outperforms those with 1k greatly.  I suppose if using vit models pretrained on 1k imagenet dataset, the gap between TransReID and other cnn-based networks could be narrowed.
Anyway, transformer benefits a lot from dataset scale which can also be considered as a trick for downstream works like person reid. If you have also tried some experiments over such pretraining issues, I would appreciate if you can share some findings with me.

**Comment by heshuting555** [comment]
Yes, Pretraining matters.
We have also done experiments on DeiT (using imagenet 1K) to make a fair comparison with other ReID methods.

Thanks for you attention!

**Comment by snowbanana** [comment]
Hi, morgenzhang!
Recently I am trying to use swin transformer to replace vit as backbone, but I discover that pretrained model in imagenet can not directly used if the person image's size is [256,128].So I resize person image to [224,224], the result doesn't look well through this
way. Can you give me some advices about how to adjust to make it work normally, thank you!

**Comment by snowbanana** [comment]
> Many thanks for you breakthrough work of introducing transformer into Person Re-id. I have tried your code locally and it works
> well as expected. However, I just find that the pretraining models from imagenet may have a significant influence on the downstream reid performance. The vit-base model was pretrained on imagenet21k and finetuned on imagenet 1k afterwards, while other cnn-based networks were mostly pretrained on imagenet 1k directly instead. So in fair, I think we have to compare these models by using the pretrained model from the same data scale.
> I have also replaced the vit with the swin architecture which provides 1k pretrained model as well as 22k. The experiments with 22k pretrained models on market1501 and msmt17 outperforms those with 1k greatly. I suppose if using vit models pretrained on 1k imagenet dataset, the gap between TransReID and other cnn-based networks could be narrowed.
> Anyway, transformer benefits a lot from dataset scale which can also be considered as a trick for downstream works like person reid. If you have also tried some experiments over such pretraining issues, I would appreciate if you can share some findings with me.

Hi, morgenzhang!
Recently I am trying to use swin transformer to replace vit as backbone, but I discover that pretrained model in imagenet can not directly used if the person image's size is [256,128].So I resize person image to [224,224], the result doesn't look well through this
way. Can you give me some advices about how to adjust to make it work normally, thank you!

**Comment by morgenzhang** [comment]
> > Many thanks for you breakthrough work of introducing transformer into Person Re-id. I have tried your code locally and it works
> > well as expected. However, I just find that the pretraining models from imagenet may have a significant influence on the downstream reid performance. The vit-base model was pretrained on imagenet21k and finetuned on imagenet 1k afterwards, while other cnn-based networks were mostly pretrained on imagenet 1k directly instead. So in fair, I think we have to compare these models by using the pretrained model from the same data scale.
> > I have also replaced the vit with the swin architecture which provides 1k pretrained model as well as 22k. The experiments with 22k pretrained models on market1501 and msmt17 outperforms those with 1k greatly. I suppose if using vit models pretrained on 1k imagenet dataset, the gap between TransReID and other cnn-based networks could be narrowed.
> > Anyway, transformer benefits a lot from dataset scale which can also be considered as a trick for downstream works like person reid. If you have also tried some experiments over such pretraining issues, I would appreciate if you can share some findings with me.
> 
> Hi, morgenzhang!
> Recently I am trying to use swin transformer to replace vit as backbone, but I discover that pretrained model in imagenet can not directly used if the person image's size is [256,128].So I resize person image to [224,224], the result doesn't look well through this
> way. Can you give me some advices about how to adjust to make it work normally, thank you!

Yes, swin transformer is indeed restricted by the input size. I also resized images to [224, 224] and slightly fine tuned the training hyperparameters based on TransReID (e.g. larger learning rate from 0.008 to 0.032 or 0.016). You can also try different architectures (tiny, small, base) and using the pretrained models on imagenet 22k, which outperforms 1k pretrained models. Actually, the results of my experiments on swin is close to TransReID.

**Comment by snowbanana** [comment]
> > > Many thanks for you breakthrough work of introducing transformer into Person Re-id. I have tried your code locally and it works
> > > well as expected. However, I just find that the pretraining models from imagenet may have a significant influence on the downstream reid performance. The vit-base model was pretrained on imagenet21k and finetuned on imagenet 1k afterwards, while other cnn-based networks were mostly pretrained on imagenet 1k directly instead. So in fair, I think we have to compare these models by using the pretrained model from the same data scale.
> > > I have also replaced the vit with the swin architecture which provides 1k pretrained model as well as 22k. The experiments with 22k pretrained models on market1501 and msmt17 outperforms those with 1k greatly. I suppose if using vit models pretrained on 1k imagenet dataset, the gap between TransReID and other cnn-based networks could be narrowed.
> > > Anyway, transformer benefits a lot from dataset scale which can also be considered as a trick for downstream works like person reid. If you have also tried some experiments over such pretraining issues, I would appreciate if you can share some findings with me.
> > 
> > 
> > Hi, morgenzhang!
> > Recently I am trying to use swin transformer to replace vit as backbone, but I discover that pretrained model in imagenet can not directly used if the person image's size is [256,128].So I resize person image to [224,224], the result doesn't look well through this
> > way. Can you give me some advices about how to adjust to make it work normally, thank you!
> 
> Yes, swin transformer is indeed restricted by the input size. I also resized images to [224, 224] and slightly fine tuned the training hyperparameters based on TransReID (e.g. larger learning rate from 0.008 to 0.032 or 0.016). You can also try different architectures (tiny, small, base) and using the pretrained models on imagenet 22k, which outperforms 1k pretrained models. Actually, the results of my experiments on swin is close to TransReID.



> > > Many thanks for you breakthrough work of introducing transformer into Person Re-id. I have tried your code locally and it works
> > > well as expected. However, I just find that the pretraining models from imagenet may have a significant influence on the downstream reid performance. The vit-base model was pretrained on imagenet21k and finetuned on imagenet 1k afterwards, while other cnn-based networks were mostly pretrained on imagenet 1k directly instead. So in fair, I think we have to compare these models by using the pretrained model from the same data scale.
> > > I have also replaced the vit with the swin architecture which provides 1k pretrained model as well as 22k. The experiments with 22k pretrained models on market1501 and msmt17 outperforms those with 1k greatly. I suppose if using vit models pretrained on 1k imagenet dataset, the gap between TransReID and other cnn-based networks could be narrowed.
> > > Anyway, transformer benefits a lot from dataset scale which can also be considered as a trick for downstream works like person reid. If you have also tried some experiments over such pretraining issues, I would appreciate if you can share some findings with me.
> > 
> > 
> > Hi, morgenzhang!
> > Recently I am trying to use swin transformer to replace vit as backbone, but I discover that pretrained model in imagenet can not directly used if the person image's size is [256,128].So I resize person image to [224,224], the result doesn't look well through this
> > way. Can you give me some advices about how to adjust to make it work normally, thank you!
> 
> Yes, swin transformer is indeed restricted by the input size. I also resized images to [224, 224] and slightly fine tuned the training hyperparameters based on TransReID (e.g. larger learning rate from 0.008 to 0.032 or 0.016). You can also try different architectures (tiny, small, base) and using the pretrained models on imagenet 22k, which outperforms 1k pretrained models. Actually, the results of my experiments on swin is close to TransReID.

Thanks~
I think it does work better after my adjustment, but seems still have little differences with TransReID. I wonder Swin transformer achieves sota performence in many fields, why the performance is just comparable to TransReID baseline when it is applying to ReID?

**Comment by michuanhaohao** [comment]
> > Many thanks for you breakthrough work of introducing transformer into Person Re-id. I have tried your code locally and it works
> > well as expected. However, I just find that the pretraining models from imagenet may have a significant influence on the downstream reid performance. The vit-base model was pretrained on imagenet21k and finetuned on imagenet 1k afterwards, while other cnn-based networks were mostly pretrained on imagenet 1k directly instead. So in fair, I think we have to compare these models by using the pretrained model from the same data scale.
> > I have also replaced the vit with the swin architecture which provides 1k pretrained model as well as 22k. The experiments with 22k pretrained models on market1501 and msmt17 outperforms those with 1k greatly. I suppose if using vit models pretrained on 1k imagenet dataset, the gap between TransReID and other cnn-based networks could be narrowed.
> > Anyway, transformer benefits a lot from dataset scale which can also be considered as a trick for downstream works like person reid. If you have also tried some experiments over such pretraining issues, I would appreciate if you can share some findings with me.
> 
> Hi, morgenzhang!
> Recently I am trying to use swin transformer to replace vit as backbone, but I discover that pretrained model in imagenet can not directly used if the person image's size is [256,128].So I resize person image to [224,224], the result doesn't look well through this
> way. Can you give me some advices about how to adjust to make it work normally, thank you!

I have tried swin transformer. For [224,224] or [256,256] image size, swin transformer (22K pretraining) achieves 89+% mAP on the Market. For [256,128] image size, swin transformer only achieves 86.8% mAP on the Market. Swin transformer is more suitable for square images.
